src/cpuinfo_x86.c

```diff
@@ -1408,18 +1425,23 @@ X86Info GetX86Info(void) {
 CacheInfo GetX86CacheInfo(void) {
   CacheInfo info = kEmptyCacheInfo;
   const Leaf leaf_0 = CpuId(0);
-  const uint32_t max_cpuid_leaf = leaf_0.eax;
-  if (IsVendor(leaf_0, "GenuineIntel")) {
-    ParseLeaf2(max_cpuid_leaf, &info);
-    ParseLeaf4(max_cpuid_leaf, &info);
+  if (IsVendor(leaf_0, CPU_FEATURES_VENDOR_GENUINE_INTEL)) {
+    ParseLeaf2(leaf_0.eax, &info);
+    ParseCacheInfo(leaf_0.eax, 4, &info);
+  } else if (IsVendor(leaf_0, CPU_FEATURES_VENDOR_AUTHENTIC_AMD) ||
+             IsVendor(leaf_0, CPU_FEATURES_VENDOR_HYGON_GENUINE)) {
+    const uint32_t max_ext = CpuId(0x80000000).eax;
+    if (!IsReservedAMD(max_ext, 22)) {
```

Comment:
22 is bit of CPUID Fn8000_0001_ECX Feature Identifiers which is TopologyExtensions and Indicates support for CPUID Fn8000_001D_EAX_x[N:0]-CPUID Fn8000_001E_EDX.